Libplanet/Store/TrieStateStore.cs

```diff
@@ -92,6 +93,51 @@ public void PruneStates(IImmutableSet<HashDigest<SHA256>> survivingStateRootHash
             stopwatch.Stop();
         }
 
+        /// <summary>
+        /// Copies states under state root hashes of given <paramref name="stateRootHashes"/>
+        /// to <paramref name="targetStateStore"/>.
+        /// </summary>
+        /// <param name="stateRootHashes">The state root hashes of states to copy.</param>
+        /// <param name="targetStateStore">The target state store to copy state root hashes.</param>
+        public void CopyStates(
+            IImmutableSet<HashDigest<SHA256>> stateRootHashes, TrieStateStore targetStateStore)
+        {
+            // TODO: As MerkleTrie now have two levels of Merkle trees (one for accounts and one for
+            // Bencodex values), it needs to be fixed so that it can prune offloaded Bencodex
+            // values too.  https://github.com/planetarium/libplanet/issues/1653
+            IKeyValueStore targetKeyValueStore = targetStateStore._stateKeyValueStore;
+            var stopwatch = new Stopwatch();
+            _logger.Verbose($"Started {nameof(CopyStates)}()");
+            var codec = new Codec();
+            foreach (HashDigest<SHA256> stateRootHash in stateRootHashes)
+            {
+                var stateTrie = new MerkleTrie(
+                    _stateKeyValueStore,
+                    new HashNode(stateRootHash),
+                    _secure
+                );
+                _logger.Debug("Started to copy states.");
+                stopwatch.Start();
+
+                foreach (var (key, value) in stateTrie.IterateNodeKeyValuePairs())
+                {
+                    targetKeyValueStore.Set(key, value);
+                }
+
+                _logger.Debug(
+                    "Finished to copy states (elapsed: {ElapsedMilliseconds} ms).",
+                    stopwatch.ElapsedMilliseconds);
+                stopwatch.Stop();
+            }
+
+            _logger.Debug(
+                "Finished to copy all states {ElapsedMilliseconds} ms",
+                stopwatch.ElapsedMilliseconds);
+            stopwatch.Stop();
```

Comment:
When does this stopwatch start?  It seems that `.Start()` call is made inside the above `foreach` block, but it corresponds to `.Stop()` call inside the same block.

Reply:
I'm sorry about it. It was because I copied `TrieStateStore.PruneStates` and cropped parts of it, and I made the mistake. I made it check only the entire loop in b9a7c52.